### PR TITLE
add teams endpoint

### PR DESCRIPTION
**Pull request by jasonlukeball**
#### Why?
Adds support for the `/teams` endpoint released in API v 1.4 so folks can get a list of teams using `intercom.teams.all` ([API changelog](https://developers.intercom.com/building-apps/docs/api-changelog)) 

#### How?
I basically copied the `/admins` endpoint ;)

### CODE REVIEW

File: lib/intercom/team.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'intercom/traits/api_resource'
+
+module Intercom
+  class Team
+    include Traits::ApiResource
+  end
+end
```
Comment (jonnyom) [inline lib/intercom/team.rb]:
```suggestion
end

```

File: lib/intercom/service/team.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      include ApiOperations::List
+      include ApiOperations::Find
+
+      def collection_class
+        Intercom::Team
+      end
+
+    end
+  end
+end
```
Comment (jonnyom) [inline lib/intercom/service/team.rb]:
```suggestion
end

```

File: spec/unit/intercom/team_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  it 'gets an team list' do
+    client.expects(:get).with("/teams", {}).returns(test_team_list)
+    client.teams.all.each { |t| }
+  end
+
+  it "gets an team" do
+    client.expects(:get).with("/teams/1234", {}).returns(test_team)
+    client.teams.find(:id => "1234")
+  end
+end
```
Comment (jonnyom) [inline spec/unit/intercom/team_spec.rb]:
```suggestion
end

```

File: lib/intercom/team.rb (hunk 1 above)
Comment (jasonlukeball) [reply lib/intercom/team.rb]: added in the new commit!

File: lib/intercom/service/team.rb (hunk 2 above)
Comment (jasonlukeball) [reply lib/intercom/service/team.rb]: added in the new commit!

File: spec/unit/intercom/team_spec.rb (hunk 3 above)
Comment (jasonlukeball) [reply spec/unit/intercom/team_spec.rb]: added in the new commit!

**Pull request merged**